lib/manageiq/messaging/kafka/client.rb

```diff
@@ -53,27 +43,27 @@ class Client < ManageIQ::Messaging::Client
         include BackgroundJob
         include Topic
 
-        private *delegate(:subscribe, :unsubscribe, :publish, :to => :kafka_client)
-        delegate :close, :to => :kafka_client
-
         attr_accessor :encoding
 
         def ack(ack_ref)
-          @queue_consumer.try(:mark_message_as_processed, ack_ref)
-          @topic_consumer.try(:mark_message_as_processed, ack_ref)
+          ack_ref.commit
+        rescue Rdkafka::RdkafkaError => e
+          logger.warn("ack failed with error #{e.message}")
+          raise unless e.message =~ /no_offset/
```

Comment:
Is a `no_offset` error something that no client would ever want an exception for from `.commit`?

Reply:
> You can generally handle the NO_OFFSETS error as a success-indicator, the offsets are already committed.

From https://github.com/confluentinc/confluent-kafka-python/issues/295 so this looks fine then

Reply:
I found this error when we tried to ack a message multiple times. I think it should ignore this error.